### PR TITLE
Add C implementation for `@stdlib/math/base/special/asin`

### LINKED ISSUE #648: [RFC]: Add C implementation for `@stdlib/math/base/special/asin`
### Description

This RFC proposes adding a C implementation for [`@stdlib/math/base/special/asin`](https://github.com/stdlib-js/stdlib/blob/b43a75923e47775446d3b6d7219ee42e07f7ce63/lib/node_modules/%40stdlib/math/base/special/asin).

```c
double stdlib_base_asin( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #648.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] evalpoly.js
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/asin/scripts/evalrational.js [hunk 1]
```diff
@@ -24,10 +24,15 @@
...
 var resolve = require( 'path' ).resolve;
+var readFileSync = require( '@stdlib/fs/read-file' ).sync;
 var writeFileSync = require( '@stdlib/fs/write-file' ).sync;
 var currentYear = require( '@stdlib/time/current-year' );
+var substringBefore = require( '@stdlib/string/substring-before' );
+var substringAfter = require( '@stdlib/string/substring-after' );
+var format = require( '@stdlib/string/format' );
 var licenseHeader = require( '@stdlib/_tools/licenses/header' );
 var compile = require( '@stdlib/math/base/tools/evalrational-compile' );
+var compileC = require( '@stdlib/math/base/tools/evalpoly-compile-c' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/scripts/evalrational.js]: You need `evalrational-compile-c` here which is on the latest `develop`.

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/special/asin/scripts/evalrational.js]: Thanks! I was gonna ask to check it :)

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a final review.

**Comment by Pranavchiku** [comment]
@kgryte when `evalrational.js` generates c-functions it defines them as 
```c
static double rational_rs( const double x )
```
and then the const variable `x` is used in the assignment, which throws an error while installing node addons.

File: lib/node_modules/@stdlib/math/base/special/asin/src/main.c [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	if ( x < 0.0 ) {
+		ax = -x;
+	} else {
+		ax = x;
+	}
+	if ( ax <= 1.0 ) {
+		s1 = 28.536655482610616 + (x * (-25.56901049652825 + (x * (6.968710824104713 + (x * (-0.5634242780008963 + (x * 0.002967721961301243)))))));
+		s2 = 342.43986579130785 + (x * (-383.8770957603691 + (x * (147.0656354026815 + (x * (-21.947795316429207 + (x * 1.0)))))));
+	} else {
+		x = 1.0 / x;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asin/src/main.c:114]: Refer previous comment

**Comment by kgryte** [comment]
@Pranavchiku Thanks for finding this bug. I'll fix and push a change to `develop` shortly.

**Comment by kgryte** [comment]
@Pranavchiku This should now be fixed on the latest `develop`.

File: lib/node_modules/@stdlib/math/base/special/asin/README.md [hunk 3]
```diff
@@ -75,6 +75,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+### Usage
+
+```c
+#include "stdlib/math/base/special/asin.h"
+```
+
+#### stdlib_base_asin( x )
+
+Computes the [arcsine][arcsine] of a `number` (in radians).
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/README.md]:
```suggestion
Computes the [arcsine][arcsine] of a number (in radians).
```

File: lib/node_modules/@stdlib/math/base/special/asin/README.md [hunk 4]
```diff
@@ -75,6 +75,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+```c
+double out = stdlib_base_asin( 0.0 );
+// returns 0.0
+out = stdlib_base_asin( -3.141592653589793/6.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/README.md:112]:
```suggestion

out = stdlib_base_asin( -3.141592653589793/6.0 );
```

File: lib/node_modules/@stdlib/math/base/special/asin/README.md [hunk 5]
```diff
@@ -75,6 +75,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/asin.h"
+#include <stdlib.h>
+#include <stdio.h>
+int main() {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/README.md:147]:
```suggestion

int main() {
```

File: lib/node_modules/@stdlib/math/base/special/asin/README.md [hunk 6]
```diff
@@ -75,6 +75,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+    double x[] = { -1, -0.78, -0.56, -0.33, -0.11, 0.11, 0.33, 0.56, 0.78, 1 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/README.md]:
```suggestion
    double x[] = { -1.0, -0.78, -0.56, -0.33, -0.11, 0.11, 0.33, 0.56, 0.78, 1.0 };
```

File: lib/node_modules/@stdlib/math/base/special/asin/benchmark/c/native/benchmark.c [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+int main( void ) {
+	double elapsed;
+	int i;
+
+	// Use the current time to seed the random number generator:
+	srand( time( NULL ) );
+
+	print_version();
+	for ( i = 0; i < REPEATS; i++ ) {
+		printf( "# c::%s\n", NAME );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/benchmark/c/native/benchmark.c]:
```suggestion
		printf( "# c::native::%s\n", NAME );
```

File: lib/node_modules/@stdlib/math/base/special/asin/examples/c/example.c [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/asin.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x[] = { -1, -0.78, -0.56, -0.33, -0.11, 0.11, 0.33, 0.56, 0.78, 1 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/examples/c/example.c]:
```suggestion
    double x[] = { -1.0, -0.78, -0.56, -0.33, -0.11, 0.11, 0.33, 0.56, 0.78, 1.0 };
```

File: lib/node_modules/@stdlib/math/base/special/asin/include/stdlib/math/base/special/asin.h [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Compute the arcsine of a number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/include/stdlib/math/base/special/asin.h]:
```suggestion
* Computes the arcsine of a number.
```

File: lib/node_modules/@stdlib/math/base/special/asin/lib/native.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Computes the arcsine of a number.
+*
+* @param {number} x - input value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/lib/native.js:32]:
```suggestion
* @private
* @param {number} x - input value
```

File: lib/node_modules/@stdlib/math/base/special/asin/src/main.c (hunk 2 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/asin/src/main.c:114]: After merging, I'll rerun the script to include the latest generated code.

File: lib/node_modules/@stdlib/math/base/special/asin/src/main.c [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+	}
+	return s1 / s2;
+}
+
+// END: rational_rs
+
+/* End auto-generated functions. */
+
+/**
+* Compute the arcsine of a number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/asin/src/main.c]:
```suggestion
* Computes the arcsine of a number.
```

**Review by kgryte — APPROVED** [review]

**Pull request merged**